client/src/AppRoot.tsx

```diff
@@ -36,49 +37,72 @@ const AppRoot = () => {
     history.push(routerPaths.root)
   }
 
-  const addElection = (newElection: MailElection) => {
-    const newElections = (mailElections || []).concat([newElection])
-    setMailElections(newElections)
-  }
+  const updateMailElections = useCallback(async () => {
+    try {
+      const { mailElections } = await getMailElections()
+      setMailElections(mailElections)
+    } catch (error) {
+      history.push(routerPaths.root)
+      window.location.reload(false)
+      console.error('getAndSetMailElections failed', error) // eslint-disable-line no-console
+    }
+  }, [history])
+
+  useEffect(() => {
+    updateMailElections()
+  }, [
+    setMailElections,
+    updateMailElections,
+    user,
+  ])
+
+  const updateVoters = useCallback(async () => {
+    mailElections?.forEach(async me => {
+      const { voters: newVoters } = await getVoters({ electionId: me.id })
+      setVoters(v => ({
+        ...v,
+        [me.id]: newVoters,
+      }))
+    })
+  }, [mailElections])
 
   useEffect(() => {
-    const getAndSetMailElections = async () => {
+    updateVoters()
+  }, [setVoters, mailElections, updateVoters])
+
+  const updateElectionDefinitions = useCallback(async () => {
+    mailElections?.map(async me => {
```

Comment:
same comment here: I don't see a good reason to load all the definitions at listing time, only when a single election is loaded at the earliest, maybe not even until the definition itself is viewed.

Reply:
Yes, will fix in future update.